text/0000-run-callback.md

```diff
@@ -0,0 +1,48 @@
+- Start Date: 2016-1-28
+- RFC PR:
+- Ember Issue:
+
+# Summary
+
+Add `Ember.run.callback()` to support running asynchronous code which tests (specifically `andThen()`) will wait for before considering the app "settled".
+
+# Motivation
+
+Ember's asynchronous test helpers (i.e. `visit()`, `fillIn()`, `andThen()`) provide a robust approach to acceptance testing in the face of asynchronous user interactions and application code. The `andThen()` helper is the backbone of this approach - it waits until the app has "settled" (i.e. async activity has competed) before asserting the desired state.
+
+For many, if not most, ambitious web applications, integration with third party libraries that require asynchronous interaction is a requirement. But unfortunately, there is currently no elegant solution for informing Ember of this asynchronous activity without resorting to lower level APIs like `registerWaiter()`.
+
+For such a common problem, Ember's opinionated approach needs an opinionated answer that matches the elegance of the rest of the test API.
+
+# Detailed design
+
+This RFC adds a single method to the `Ember.run` namespace: `Ember.run.callback()` (better naming suggestions are welcome!).
+
+The method takes a single function as an argument, and returns a wrapper function that can be used in place of the originally supplied one. This wrapper does two things:
+
+1. Ensures the callback is wrapped in an `Ember.run()` call, mostly for convenience.
+2. More importantly, it wraps the user-supplied function with a `registerWaiter` method (and the associated state management / cleanup) which resolves once the
```

Comment:
That's fair. My thought was that the behavior here matches similar behavior with the other Ember.run.* methods (i.e. want an Ember aware setTimeout? `run.later`. Want an Ember aware callback? `run.callback`). It mixes concerns under the hood, but from my perspective when trying to solve this problem, `Ember.run` was my gut reaction first place to check.

Certainly open to suggestions of a better spot for it to live.